drivers/tee/optee/optee.c

```diff
@@ -449,6 +512,38 @@ static void free_shm_pages(void **pages)
 	}
 }
 
+static void handle_rpc_supp_cmd(const struct device *dev, struct optee_msg_arg *arg)
+{
+	struct tee_param *params;
+	int ret;
+
+	arg->ret_origin = TEEC_ORIGIN_COMMS;
+
+	params = k_malloc(sizeof(*params) * arg->num_params);
+	if (!params) {
+		arg->ret = TEEC_ERROR_OUT_OF_MEMORY;
+		return;
+	}
+
+	ret = msg_param_to_param(params, arg->num_params, arg->params);
+	if (ret) {
+		arg->ret = TEEC_ERROR_BAD_PARAMETERS;
+		arg->ret_origin = TEEC_ORIGIN_COMMS;
+		goto out;
+	}
+
+	arg->ret = optee_call_supp(dev, arg->cmd, arg->num_params, params);
+
+	ret = param_to_msg_param(params, arg->num_params, arg->params);
+	if (ret) {
+		arg->ret = TEEC_ERROR_GENERIC;
+		arg->ret_origin = TEEC_ORIGIN_COMMS;
+		goto out;
```

Comment:
nit: no need for this goto

Reply:
someone can add `goto` if he or she add a new chunk of code